proposals/format.md

```diff
@@ -0,0 +1,320 @@
+the compiler won't emit a call to it.
+
+### CLR stack allocation helpers
+The CLR only provides only 
+[localloc](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.emit.opcodes.localloc?redirectedfrom=MSDN&view=netframework-4.7.2)
+ for stack allocation of contiguous memory. This instruction is limited in that it it only works for `unmanaged` types. 
+ This means it can't be used as a universal solution for efficientally allocating the backing storage for `params 
+ Span<T>`. 
+
+This limitation is not some fundamental restriction though but instead more an artifact of history. The CLR could choose
+to add new op codes / intrinsicts which provide universal stack allocation. These could then be used to allocate the
+backing storage for most `params Span<T>` calls.
+
+``` csharp
+static class BetterAllocation {
+    static void Use(params Span<string> spans) {
+        ...
+    }
+
+    static void Go() {
+        Use("hello", "world");
+    }
+}
+```
+
+The `Go` method can be lowered to the following:
+
+``` csharp
+static class ZeroAllocation {
+    static void Go() {
+        Span<T> span = RuntimeIntrinsic.StackAlloc<string>(length: 2);
+        span[0] = "hello";
+        span[1] = "world";
+        Use(span);
+    }
+}
+```
+
+While this approach is very heap efficient it does cause extra stack usage. In an algorthim which has a deep stack and
+lots of `params` usage it's possible this could cause a `StackOverflowException` to be generated where a simple `T[]`
+allocation would succeed. 
+
+Unfortunately C# is not set up for the type of inter-method analysis where it could make an educated determination of
+whether or not call should use stack or heap allocation of `params`. It can only really consider each call on its 
+own.
+
+The CLR is best setup for making this type of determination at runtime. Hence we'd likely have the runtime provide two
+methods for universal stack allocation:
+
+1. `Span<T> StackAlloc<T>(int length)`: this has the same behaviors and limitations of `localloc` except it can work on
+any type `T`. 
+1. `Span<T> MaybeStackAlloc<T>(int length)`: this runtime can choose to implement this by doing a stack or heap
+allocation. The runtime can then use the execution context in which it's called to determine how the `Span<T>` is 
+allocated. The caller though will always treat it as if it were stack allocated.
+
+For very simple cases, like one to two arguments, the C# compiler could always use `StackAlloc<T>` variant. This is 
+unlikely to significantly contribute to stack exhaustion in most cases. For other cases the compiler could choose to 
+use `MaybeStackAlloc<T>` instead and let the runtime make the call.
+
+How we choose will likely require a deeper investigation and examination of real world apps. But if these new intrinsics
+are available then it will give us this type of flexibility.
+
+### Why not varargs? 
+!!! Will flesh this out in a bit. Have to track down my notes!!!
```

Comment:
FYI: i did not forget about this. I'm actually trying to track down my notes. Verdict though is it's not a good solution for C#.